File: Navidad 2024/1. First gift repeated!/main.js
function prepareGifts(gifts) {
    let allGifts = [];
    let aux = 0;

    for (let i = 0; i < gifts.length; i++) {
        if (!allGifts.includes(gifts[i])) {
            allGifts.push(gifts[i]);
        } else {
            continue;
        }
    }

    for (let i = 0; i < allGifts.length; i++) {
        for (let j = 0; j < allGifts.length; j++) {
            if (allGifts[j] > allGifts[j + 1]) {
                aux = allGifts[j];
                allGifts[j] = allGifts[j + 1];
                allGifts[j + 1] = aux;
            }
        }
    }

    return allGifts;
}

File: Navidad 2024/1. First gift repeated!/main.py
def prepare_gifts(gifts):
    # Declaramos variables
    allGifts = []
    aux = 0
    n = len(gifts)

    # Bucle que hará el ordenamiento burbuja sencillo
    for i in range(n):
        for j in range(n - 1):
            if gifts[j] > gifts[j + 1]:
                aux = gifts[j]
                gifts[j] = gifts[j + 1]
                gifts[j + 1] = aux

        # Comprobación para agregar el elemento NO repetido al arreglo
        if gifts[i] not in allGifts:
            allGifts.append(gifts[i])

    return allGifts


print(prepare_gifts([]))
